src/pages/index.js

```diff
@@ -67,79 +69,123 @@ const validationSettings = {
   errorClass: "modal__error_visible",
 };
 
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "298dccb5-56b1-425a-ba81-960441ee84bf",
+    "Content-Type": "application/json",
+  },
+});
+
+const avatarCardPopUp = new PopUpWithForm(
+  "#edit-avatar-modal",
+  handleAvatarSubmit
+);
+
+avatarCardPopUp.setEventListeners();
+
 const editFormValidator = new FormValidator(
   validationSettings,
   profileEditForm
 );
 const addFormValidator = new FormValidator(validationSettings, addCardForm);
+const updateAvatarValidator = new FormValidator(
+  validationSettings,
+  avatarUpdateForm
+);
 
 editFormValidator.enableValidation();
 addFormValidator.enableValidation();
 editFormValidator.disableButton();
 addFormValidator.disableButton();
+updateAvatarValidator.enableValidation();
+updateAvatarValidator.disableButton();
 
 /* const closeModal = (modalWindow) => {
   modalWindow.classList.remove("modal_open");
   document.removeEventListener("keydown", handleEscapeKey);
 }; */
 
 function createCard(item) {
-  const cardElement = new Card(item, "#card-template", handleCardClick);
+  const cardElement = new Card(
+    item,
+    "#card-template",
+    handleCardClick,
+    handleDeleteClick,
+    handleLikeClick
+  );
   return cardElement.getView();
 }
 
-/* con = (data, wrap) => {
-  const card = createCard(data);
-  wrap.prepend(card);
-}; */
-
 const profileName = document.querySelector(".profile__name");
 const profileDescription = document.querySelector(".profile__description");
-
+const avatar = document.querySelector(".profile__image");
 const profileNameInput = document.querySelector("#profile-name-input");
 const profileDescriptionInput = document.querySelector(
   "#profile-description-input"
 );
 
-const userData = new UserInfo(profileName, profileDescription);
-
-function handleProfileEditSubmit() {
-  userData.setUserInfo(profileNameInput.value, profileDescriptionInput.value);
+function handleProfileEditSubmit(userProfile) {
+  popupWithEditForm.submitLoading(true);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```